pkg/roachprod/failureinjection/failures/disk_stall.go

```diff
@@ -0,0 +1,474 @@
+// Copyright 2025 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package failures
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/roachprod"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/install"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/logger"
+	"github.com/cockroachdb/errors"
+)
+
+var cockroachIOController = filepath.Join("/sys/fs/cgroup/system.slice", install.VirtualClusterLabel(install.SystemInterfaceName, 0)+".service", "io.max")
+
+type CGroupDiskStaller struct {
+	GenericFailure
+}
+
+func MakeCgroupDiskStaller(clusterName string, l *logger.Logger, secure bool) (FailureMode, error) {
+	c, err := roachprod.GetClusterFromCache(l, clusterName, install.SecureOption(secure))
+	if err != nil {
+		return nil, err
+	}
+	genericFailure := GenericFailure{c: c, runTitle: CgroupsDiskStallName}
+	return &CGroupDiskStaller{GenericFailure: genericFailure}, nil
+}
+
+const CgroupsDiskStallName = "cgroup-disk-stall"
+
+func registerCgroupDiskStall(r *FailureRegistry) {
+	r.add(CgroupsDiskStallName, DiskStallArgs{}, MakeCgroupDiskStaller)
+}
+
+func (s *CGroupDiskStaller) Description() string {
+	return CgroupsDiskStallName
+}
+
+type DiskStallArgs struct {
```

Comment:
Nit: move above the `func` s.t. all var/const/type declarations precede the definitions.